hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/store/contract/MirrorEntityAccess.java

```diff
@@ -0,0 +1,106 @@
+package com.hedera.mirror.web3.evm.store.contract;
+
+/*-
+ * ‌
+ * Hedera Mirror Node
+ * ​
+ * Copyright (C) 2019 - 2022 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import static com.hedera.mirror.common.domain.entity.EntityType.TOKEN;
+import static com.hedera.services.evm.accounts.HederaEvmContractAliases.isMirror;
+
+import com.google.protobuf.ByteString;
+import java.util.Optional;
+import javax.inject.Named;
+import lombok.RequiredArgsConstructor;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.units.bigints.UInt256;
+import org.hyperledger.besu.datatypes.Address;
+
+import com.hedera.mirror.common.domain.entity.Entity;
+import com.hedera.mirror.common.domain.entity.EntityIdEndec;
+import com.hedera.mirror.common.util.DomainUtils;
+import com.hedera.mirror.web3.repository.ContractRepository;
+import com.hedera.mirror.web3.repository.ContractStateRepository;
+import com.hedera.mirror.web3.repository.EntityAccessRepository;
+import com.hedera.services.evm.store.contracts.HederaEvmEntityAccess;
+
+@Named
+@RequiredArgsConstructor
+public class MirrorEntityAccess implements HederaEvmEntityAccess {
+    private final EntityAccessRepository entityRepository;
+    private final ContractRepository contractRepository;
+    private final ContractStateRepository contractStateRepository;
+
+    @Override
+    public boolean isUsable(Address address) {
+        //TODO impl
+        return false;
+    }
+
+    @Override
+    public long getBalance(Address address) {
+        final var entity = findEntity(address);
+        return entity.isPresent() ? entity.get().getBalance() : 0L;
+    }
+
+    @Override
+    public boolean isExtant(Address address) {
+        return entityRepository.existsById(entityIdFromEvmAddress(address));
+    }
+
+    @Override
+    public boolean isTokenAccount(Address address) {
+        final var entity = findEntity(address);
+        return entity.isPresent() && entity.get().getType().equals(TOKEN);
+    }
+
+    @Override
+    public ByteString alias(Address address) {
+        final var entity = findEntity(address);
+        return entity.map(value -> ByteString.copyFrom(value.getAlias())).orElse(ByteString.EMPTY);
```

Comment:
Bit strange to return proto back. But we should use the util to avoid an allocation.
```suggestion
        return entity.map(value -> DomainUtils.fromBytes(value.getAlias())).orElse(ByteString.EMPTY);
```